client/app/cloud/project/compute/loadbalancer/configure/compute-loadbalancer-configure.controller.js

```diff
@@ -0,0 +1,260 @@
+        }).finally(() => self.loaders.table.server = false);
+    }
+
+    // Configure and deploy the loadbalancer
+    self.configure = function () {
+        if (self.loaders.form.loadbalancer) {
+            return;
+        }
+        self.loaders.form.loadbalancer = true;
+        var promise = $q.resolve("");
+
+        // Configure the HTTP(80) loadbalancer
+        var configLoadBalancer = _.values(self.form.servers).length && _.reduce(self.form.servers, (res,value) => res && value, true) || _.values(self.attachedServers).length > 0
+        if (self.loadbalancer.status !== "custom" && configLoadBalancer) {
+            if(self.loadbalancer.status === "available") {
+                // Create farm and front
+                promise = promise.then(() => OvhApiIpLoadBalancing.Farm().Http().Lexi().post({ serviceName : loadbalancerId}, {
+                   displayName : `PublicCloud-${serviceName}`,
+                   port : 80,
+                   zone : "all",
+               }).$promise.then((farm) => self.loadbalancer.farm = farm));
+                promise = promise.then(() => OvhApiIpLoadBalancing.Frontend().Http().Lexi().post({ serviceName : loadbalancerId}, {
+                   displayName : `PublicCloud-${serviceName}`,
+                   port : 80,
+                   zone : "all",
+                   defaultFarmId : self.loadbalancer.farm.farmId,
+               }).$promise.then((frontend) => self.loadbalancer.frontend = frontend));
+            }
+
+            // Add or remove servers
+            var modified = false;
+            _.forEach(self.form.servers, (enable, ip) => {
+                const server = _.find(self.table.server, { ip : ip })
+                const displayName = server ? server.label : null;
+                if (enable && !self.attachedServers[ip]) {
+                    modified = true;
+                    promise = promise.then(() => OvhApiIpLoadBalancing.Farm().Http().Server().Lexi().post({ serviceName : loadbalancerId, farmId : self.loadbalancer.farm.farmId}, {
+                       displayName : displayName,
+                       port : 80,
+                       address : ip,
+                       status : "active",
+                   }).$promise);
+                }
+                if (!enable && self.attachedServers[ip]) {
+                    modified = true;
+                    promise = promise.then(() => OvhApiIpLoadBalancing.Farm().Http().Server().Lexi().delete({
+                       serviceName : loadbalancerId,
+                       serverId : self.attachedServers[ip].serverId,
+                       farmId : self.loadbalancer.farm.farmId,
+                   }).$promise);
+                }
+            })
+
+            // Deploy configuration
+            if (modified) {
+                promise = promise.then(() => OvhApiIpLoadBalancing.Lexi().refresh({ serviceName : loadbalancerId }, {}).$promise);
```

Comment:
why are you re-assigning so often promise?

Reply:
promise.
    .then()
    .then()
    .then() would be better ;»)